internal/provider/resource_connector_aws.go

```diff
@@ -0,0 +1,484 @@
+				ValidateDiagFunc: validation.ToDiagFunc(
+					validation.StringIsJSON,
+				),
+			},
+		},
+		// auth_params requires a resource recreation as they cannot be updated.
+		// to accommodate for importing resources into state, we can't use `ForceNew` in the schema definition.
+		// we use a customdiff and `ForceNewIfChange` for below attributes for only a change condition.
+		CustomizeDiff: customdiff.All(
+			customdiff.ForceNewIfChange("auth_params", func(ctx context.Context, old, new, meta any) bool {
+				if old.(string) != "" {
+					return old.(string) != new.(string)
+				}
+				return false
+			},
+			),
+		),
+		CreateContext: resourceWizConnectorCreate,
+		ReadContext:   resourceWizConnectorRead,
+		UpdateContext: resourceWizConnectorUpdate,
+		DeleteContext: resourceWizConnectorDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// CreateConnector struct
+type CreateConnector struct {
+	CreateConnector wiz.CreateConnectorPayload `json:"createConnector"`
+}
+
+func resourceWizConnectorCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	tflog.Info(ctx, "resourceWizConnectorCreate called...")
+
+	query := `mutation CreateConnector($input: CreateConnectorInput!) {
+				createConnector(input: $input) {
+					connector {
+						id
+					}
+				}
+			  }
+		     `
+	// populate the graphql variables
+	vars := &wiz.CreateConnectorInput{}
+	vars.Name = d.Get("name").(string)
+	enabled := d.Get("enabled").(bool)
+	vars.Type = "aws"
+	vars.Enabled = &enabled
+
+	vars.AuthParams = json.RawMessage(d.Get("auth_params").(string))
+	vars.ExtraConfig = json.RawMessage(d.Get("extra_config").(string))
+
+	// process the request
+	data := &CreateConnector{}
+	requestDiags := client.ProcessRequest(ctx, m, vars, data, query, "connector", "create")
+	diags = append(diags, requestDiags...)
+	if len(diags) > 0 {
+		return diags
+	}
+
+	// set the id
+	d.SetId(data.CreateConnector.Connector.ID)
+
+	return resourceWizConnectorRead(ctx, d, m)
+}
+
+// ReadConnectorPayload struct
+type ReadConnectorPayload struct {
+	Connector wiz.Connector `json:"connector"`
+}
+
+func resourceWizConnectorRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	tflog.Info(ctx, "resourceWizConnectorRead called...")
+
+	// check the id
+	if d.Id() == "" {
+		return nil
+	}
+
+	// define the graphql query
+	query := `query GetConnector($id: ID!) {
+	    connector(id: $id) {
+	      id
+	      name
+	      enabled
+	      authParams
+	      extraConfig
+	      outpost {
+	        id
+	        name
+	        status
+	        config {
+	          ... on OutpostAWSConfig{
+	            roleARN
+	            externalID
+	            accessKey
+	            secretKey
+	            stateBucketName
+	            resultsBucketName
+	            settingsRegion
+	            disableNatGateway
+	            }
+	        }
+	    }
```

Comment:
When managing Connector resources, only the ID is required for the provider.  Is there a reason we're chasing the outpost details on read?  I'm trying to understand the value or need for this.  Let me know if I am missing something.

Reply:
The assumption was the outpost orchestrator role arn would be useful as an output for any IAM-type automations. Investigating a bit I don't see references to it in any IAM trust policies, so fair comment, this is probably not worth chasing.